### ISSUE TITLE
Inconsistent Dispatchers.IO visibility

### ISSUE THREAD

**Issue by jbruchanov** [Kotlin/kotlinx.coroutines#3923]
Having a multiplatform project android/jvm/ios.

I'm getting build time error on macos for the project, iOS part
```
Cannot access 'IO': it is internal in 'Dispatchers'
> Task :shared:compileKotlinIosSimulatorArm64 FAILED
```

Using coroutines 1.7.3 and IDE on macOS shows me this
![image](https://github.com/Kotlin/kotlinx.coroutines/assets/1651939/3d827995-3c18-4e11-b982-4a8d4ddd0e1c)

IDE on windows for a jvm dependency
`org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.7.3`
has it simply public.

looks like if I import this, 
```
import kotlinx.coroutines.IO
```
it works as a workaround in general, though windows IDE says it's unused so will be probably removed for import optimisation and will break the build essentially 

Windows IDE, in jvm lib I can see this...
```
@Suppress("EXTENSION_SHADOWED_BY_MEMBER")
@Deprecated(message = "Should not be used directly", level = DeprecationLevel.HIDDEN)
public actual val Dispatchers.IO: CoroutineDispatcher get() = Dispatchers.IO
```

Do I have something wrong in my project setup in terms of dependencies or is it a bug ?
Or what's the correct way of referencing Dispatchers in KMP project ?

my build.gradle
```
...
sourceSets {
    commonMain {
        dependencies {
           //kotlinx-coroutines = "1.7.3"
           //kotlinx-coroutines-core = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core", version.ref = "kotlinx-coroutines" }
            implementation(libs.kotlinx.coroutines.core)
        }
    }
}

**Comment by dkhalanskyjb** [comment]
Could you show the specific place in *your* code (not our own library) that causes an error, along with the list of imports in that file?

**Comment by jbruchanov** [comment]
@dkhalanskyjb 
Hi, 

code on Windows..
![image](https://github.com/Kotlin/kotlinx.coroutines/assets/1651939/7d2cd205-c4c0-4929-8dbe-5f06557de27f)
all good, runs in IDE, `./gradlew shared:build` works with no issue...

same code mac os
![image](https://github.com/Kotlin/kotlinx.coroutines/assets/1651939/7c7fa7f1-5200-42e4-a017-7d77fe0b0b96)

it runs in IDE, but `./gradlew shared:build` fails with
```
> Task :shared:compileCommonMainKotlinMetadata FAILED
e: ....shared/src/commonMain/kotlin/testapp/Test.kt:14:29 Unresolved reference: IO
```
If I let IDE fix it, then it adds extra import (macos)
and `./gradlew shared:build` works fine
![image](https://github.com/Kotlin/kotlinx.coroutines/assets/1651939/5d9d1c8e-f7d9-41f0-8d2c-f8a94d011820)

Then if I go back to windows it, the new import is unused
![image](https://github.com/Kotlin/kotlinx.coroutines/assets/1651939/e830fc0c-ff75-4668-8c00-fbcdd06b809c)
And optimise imports remove it so it breaks it on mac os

**Comment by jbruchanov** [comment]
MacOs
```
IntelliJ IDEA 2023.2.3 (Ultimate Edition)
Build #IU-232.10072.27, built on October 11, 2023
Licensed to Jiri Bruchanov
You have a perpetual fallback license for this version.
Subscription is active until August 9, 2024.
Runtime version: 17.0.8.1+7-b1000.32 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 10
Metal Rendering is ON
Registry:
    ide.experimental.ui=true

Non-Bundled Plugins:
    org.jetbrains.compose.desktop.ide (1.5.3)

Kotlin: 232-1.9.0-IJ10072.27
```

Windows:
```
IntelliJ IDEA 2023.2.4 (Ultimate Edition)
Build #IU-232.10203.10, built on October 24, 2023
Licensed to Jiri Bruchanov
You have a perpetual fallback license for this version.
Subscription is active until August 9, 2024.
Runtime version: 17.0.8.1+7-b1000.32 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 8192M
Cores: 32
Registry:
    debugger.new.tool.window.layout=true
    ide.experimental.ui=true

Non-Bundled Plugins:
    com.intellij.nativeDebug (232.10203.2)
    com.jetbrains.packagesearch.intellij-plugin (232.9921.28)
    org.jetbrains.compose.desktop.ide (1.5.3)
    ru.ztrap.plugin.idea.compose-color-preview (0.0.1)
    androidx.compose.plugins.idea (232.9559.10)
    com.squareup.sqldelight (2.0.0)
    Pythonid (232.10203.10)
    izhangzhihao.rainbow.brackets (2023.3.6)

Kotlin: 232-1.9.0-IJ10203.10
```

**Comment by dkhalanskyjb** [comment]
The advice is to use a star import: `import kotlinx.coroutines.*`. Would that work for you?

**Comment by jbruchanov** [comment]
@dkhalanskyjb 
Hi, not really, we disallowed using wildcards on imports...

Is this some intentional design or what's behind this difference ?
I'd assume that class signatures should be same cross platform

**Comment by dkhalanskyjb** [comment]
It's not intentional; it's just a technical limitation. Initially, only the `Dispatchers.IO` on the JVM was introduced, and there, `IO` was part of the `Dispatchers` object, so now we have to keep `IO` inside the object for backward compatibility. Later, `IO` was also added to the Native implementation. We couldn't just add `IO` to `Dispatchers` in the common code, as JS doesn't provide `IO`, so we settled for a workaround that introduced `IO` as an extension function on `Dispatchers` in the common code.

So, yes, the list of imports on Native or Native + JVM differs from the list of imports on just the JVM.
